### DISCUSSION TITLE
How do I enable conditional prompts?

### DISCUSSION THREAD

**Discussion by sQVe** [olimorris/codecompanion.nvim#264] [Q&A] [answered]
👋🏼,

I was previously using the `actions` field when configuring CodeCompanion, and I'm now trying to use the new `prompt_library`, but my `condition` functions are not being validated.

```lua
prompt_library = {
  ['Condense text'] = M.condense_text = {
    condition = function(context)
      return context.is_visual
    end,
    opts = { auto_submit = true },
    strategy = 'chat',
    prompts = {
      {
        role = 'system',
        content = format_lines({
          'Your task is to condense the text.',
          'Strive for simple, clear language, avoiding complexity or ambiguity.',
          'Start your response with the condensed text in a code block.',
          'End your response with a concise list of the changes made.',
        }),
      },
      {
        role = 'user',
        content = function(context)
          return format_lines({
            'I have the following text in `%s`:',
          }, path.get_basename(buffer.get_path(context.bufnr)))
        end,
      },
    },
  }
}
```

What am I missing? 🤔

**Comment by olimorris** [comment]
I may have missed this when I made the changes so I'll add this as an issue.

**Reply by olimorris** [reply]
Actually, turns out this should be working after all as I have this in the `Open chats` prompt. Definitely remove the `M.condense_text =` from your config above. But I'd also recommend using `opts.modes` instead of the condition. I cover this in the RECIPES.md guide.

**Reply by sQVe** [reply]
@olimorris The `M.condense_text` was a copy & paste mistake on my end when writing this issue. My configuration does not have this issue, since it's just referencing a module method called `condense_text`. 

It looks like this:
```lua
M.condense_text = {
  condition = function(context)
    return context.is_visual
  end,
  opts = { auto_submit = true },
  strategy = 'chat',
  prompts = {
    {
      role = 'system',
      content = format_lines({
        'Your task is to condense the text.',
        'Strive for simple, clear language, avoiding complexity or ambiguity.',
        'Start your response with the condensed text in a code block.',
        'End your response with a concise list of the changes made.',
      }),
    },
    {
      role = 'user',
      content = function(context)
        return format_lines({
          'I have the following text in `%s`:',
        }, path.get_basename(buffer.get_path(context.bufnr)))
      end,
    },
  },
}
```

Using `opts.modes` is not granular enough since some of my conditional prompts are very specific, like:
```lua
  condition = function(context)
    local git_diff = vim.fn.system({ 'git', 'diff', 'main..HEAD' })

    return git_diff:match('^diff')
  end,
```

I'll take a look and see if I can find where the issue is.

**Reply by sQVe** [reply, marked as answer]
@olimorris I found the issue. See https://github.com/olimorris/codecompanion.nvim/pull/277 for a fix.

The plugin is using a `condition`, but since it's a static action, it is not handled the same way as the `prompt_library`.
